### DISCUSSION TITLE
The acl of protected file in s3 is inherit bucket, is not expect private acl

### DISCUSSION THREAD

**Discussion by shynome** [pocketbase/pocketbase#3077] [Q&A] [answered]
I have a s3 bucket, it is public read and private write, I export it to end user directly for speed

I found this new feature `file protected`, and have a test is it protect the file in a public read s3 bucket, I found  it is not, 
the acl of file is inherit bucket, so the file is public read, anyone can read the protected file from s3 endpoint

At last, Thanks for your good framework, I can easy implement this thought based on your good program design

----
updated: I create a library to do it, if you have the same demand, you can have a try 
https://github.com/shynome/pocketbase-s3

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure I follow.

The [protected files](https://pocketbase.io/docs/files-handling/#protected-files) works only when using the PocketBase files APIs. It doesn't attempt to modify the S3 bucket acl policy because not all S3 vendors support the same access controls and policies management.

PocketBase interacts with the S3 storage only via the S3 API and its configured credentials.

If you want to interact directly with the S3 endpoint then it is up to the developers to manage the permissions of the bucket.

**Reply by shynome** [reply]
> because not all S3 vendors support the same access controls and policies management.

Thanks your reply, I don't know those before now got it, I will implement it in my program

**Reply by ganigeorgiev** [reply]
Note also that the protected files access is based on the View API rule of the collection, aka. it is slightly more complex than just public-private flag.

**Reply by shynome** [reply]
Thanks your tip, I will take care it. 
Before now I wrote a separate api, the api recevie the file field value then return s3 private bucket file SignedURL
